referee.c

```diff
@@ -0,0 +1,180 @@
+#include <stdio.h>
+#include <time.h>
+#include <Windows.h>
+#include <stdlib.h>
+#include "referee.h"
+
+//#define N 3
+const char markerX = 'X';
+const char marker0 ='0';
+const char markerEmpty = ' ';
+
+
+int referee(int (*playerX)(char marker, char* board_copy), int (*player0)(char marker, char* board_copy) ){
+    int movesCounter = 0;
+    char marker;  //who is playing now: x or 0
+    char board[N*N];
+    char board_copy[N*N];
+    int coord;  //coordinate of moving
+
+    initializeBoard(board);
+    int (*player)(char, char*);
+
+    showBoard(board);
+    while ( movesCounter < N*N ){
+        //showBoard(board);
+        if ((movesCounter%2) == 0){
+            marker = markerX;
+        }
+        else
+            marker = marker0;
+
+        whoMoves(marker);
+        copyBoard(board, board_copy);
+        Sleep(100);
+        if (marker == markerX){
+            player = playerX;
+        }
+        else
+           player = player0;
+        int countboner = 0;   //count faulty moves
+        int put;  // result of putMarker (0 - marker put on board, -1 - wrong move)
+        int attempt = 2; //how much player can do faulty moves
+        do {
+            coord = player(marker, board_copy);
+            put = putMarker(board, marker, coord);
+            if ( put == -1 ){
+                ++countboner;
+                (countboner < attempt) ? printf("Cell is occupied. Try once again!\n") : printf("You should to read rules. Buy!\n");

```

Comment:
I believe that human player can make unlimited number of attempts to make an invalid move, but the input mechanism should not allow them and should repeat prompt until a valid move made. In case of computer player, an algorithm should never make an invalid move.

Reply:
I don't know what kind of algorithms will be in game. I supposed that any algorithm may be imperfect and endless loop can happen. And human would hardly 3 times in a row make invalid move.

Reply:
I see the idea behind human player, but should we allow mistakes for algorithms?

Reply:
Also, we shouldn't print anything here, but delegate that to graphics.